docs/writing-stories/args.md

```diff
@@ -213,6 +213,21 @@ Similarly, special formats are available for dates and colors. Date objects will
 
 Args specified through the URL will extend and override any default values of args set on the story.
 
+## Setting args from within a story
```

Comment:
@jonthenerd, one final thing, as the underlying issue relates to React and to avoid confusion, we have a mechanism in place that allows the documentation to render for specific frameworks conditionally. Would you mind just wrapping this with the following `<IfRenderer renderer='react'> ....</IfRenderer>` Turning this into:

```markdown
<IfRenderer renderer='react'>

## Setting args from within a story


Interactive components often need to be controlled by their containing component or page to respond to events, modify their state and reflect those changes in the UI. For example, when a user toggles a switch component, the switch should be checked, and the arg shown in Storybook should reflect the change. To enable this, you can use the `useArgs` API exported by `@storybook/preview-api`:

<!-- prettier-ignore-start -->

<CodeSnippets
  paths={[    
    'react/page-story-args-within-story.js.mdx',
    'react/page-story-args-within-story.ts.mdx'
  ]}
/>

<!-- prettier-ignore-end -->

</IfRenderer>

``` 


If you have any questions, see how it's done on the [install page](https://github.com/storybookjs/storybook/blob/next/docs/get-started/install.md).

Reply:
@jonniebigodes added the IFRenderer as requested. Hopefully this is the last bit? Believe this is ready for re-review.